### PR TITLE
Update `CODEOWNERS` paths: fix invalid paths

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
As part of ongoing work of enabling wildcard support for `CODEOWNERS`:
- https://github.com/Azure/azure-sdk-tools/issues/2770
- https://github.com/Azure/azure-sdk-tools/pull/5088

and enabling stricter validation:
- https://github.com/Azure/azure-sdk-tools/issues/4859

this PR:
- fixes invalid paths, to match rules explained [here](https://github.com/Azure/azure-sdk/blob/main/docs/policies/opensource.md#codeowners);
- removes `/**/tests.yml` and `/**/ci.yml`, to avoid all build failure notifications being routed to it once we enable the new regex-based, wildcard-supporting `CODEOWNERS` matcher, per: https://github.com/Azure/azure-sdk-tools/pull/5088#issuecomment-1397330147

Once this PR is merged, I will enable the new `CODEOWNERS` matcher, similar to how it was done for `net` repo by these two PRs:
- https://github.com/Azure/azure-sdk-tools/pull/5241
- https://github.com/Azure/azure-sdk-tools/pull/5240

### CODE REVIEW

File: .github/CODEOWNERS
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:9]:
```suggestion
/**                      @rickwinter @ahsonkhan @antkmsft @vhvb1989 @gearama @LarryOsterman
```

File: .github/CODEOWNERS
Comment (LarryOsterman) [inline .github/CODEOWNERS:35]:
```suggestion
/sdk/template/           @danieljurek @weshaggard @LarryOsterman @RickWinter 
```

I believe that the template SDK is owned by the C++ team as well as the ES team.

**Comment by LarryOsterman** [comment]
The change looks ok to me, but I'm not fully aware of the full story behind this PR so I'm going to defer to the ES team to approve the PR.

File: .github/CODEOWNERS
Comment (danieljurek) [reply .github/CODEOWNERS]: Template packages are intended to be used by SDK developers who want to write an SDK for a given service. They're a starting point for those developers and as such should be owned by the SDK team.

We in EngSys do use template packages from time to time to test various release scenarios. I'm not sure how Wes and I ended up as codeowners on here. At one point we were prototyping having multiple template packages for dependency release scenarios but solved that problem by bundling all of the vcpkg releases into a single PR by default.

Comment (konrad-jamrozik) [reply .github/CODEOWNERS]: @weshaggard can you chip-in?

Comment (weshaggard) [reply .github/CODEOWNERS]: I'm good with having engsys and the sdk team listed as owners here as this pipeline is generally a litmus test for bigger issues in the repo.

**Review by weshaggard — APPROVED** [review]

**Review by danieljurek — APPROVED** [review]

**Pull request merged**
